src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CollectionCodecs.java

```diff
@@ -121,14 +174,29 @@ private static JSONCodec<Object, Object> findElementCodec(
     String msg =
         String.format(
             "no codec matching (list/set) declared element type `%s`, actual value type `%s`",
-            elementType, element.getClass());
+            elementType, element.getClass().getName());
+    throw new ToCQLCodecException(element, elementType, msg);
+  }
+
+  private static JSONCodec<Object, Object> findMapValueCodec(
+      List<JSONCodec<?, ?>> valueCodecs, DataType elementType, Object element)
+      throws ToCQLCodecException {
+    for (JSONCodec<?, ?> codec : valueCodecs) {
+      if (codec.handlesJavaValue(element)) {
+        return (JSONCodec<Object, Object>) codec;
+      }
+    }
+    String msg =
+        String.format(
+            "no codec matching map declared value type `%s`, actual value type `%s`",
```

Comment:
nit: useful to include the list of codecs that was checked ?

Reply:
That could be: one problem is that of how to represent `JSONCodec`s as text -- will try something.

Reply:
Ok cool; was able to get useful List easily -- great suggestion!